app/controllers/application_controller.rb

```diff
@@ -22,12 +23,11 @@ def append_to_log(params)
   # Rails' instrumentation code, and is run after each request.
   def append_info_to_payload(payload)
     super
-    if @custom_log_items
-      payload[:custom_log_items] = @custom_log_items
-    end
-  end
 
-  # :nocov:
+    append_to_log(request_id: RequestStore.store[:request_id])
```

Comment:
I guess a difference of opinion then, I still prefer my approach.

The way I see it, storing the request id is not really the responsibility of a controller, and would be better handled as middleware, which is also how it originally gets set. It also becomes easier to write specs which is a good sign.

I still think that calling your method inside the hook method is better than create more indirection because the code related to logging will be closer together.

I've looked into the hook method and surprised that it is a private method, we should probably add a comment about it since it could break when updating Rails.